Box2D/Extension/b2VisBody.cpp

```diff
@@ -43,12 +44,21 @@ bool b2VisBody::hasAttachedTexture() const
     return m_pTexture.get() != nullptr && m_pTexture->getTextureId()>0;
 }
 
-int b2VisBody::getUniqueId() const
+void b2VisBody::setUniqueId(const int& id)
 {
+    m_nUniqueId = id;
+}
+
+int b2VisBody::getUniqueId()
+{
+    if(m_nUniqueId>=0) {
+        return m_nUniqueId;
+    }
     int index = 0;
     b2VisBody* list = (b2VisBody*) m_world->GetBodyList();
     for (b2VisBody* b = (b2VisBody*) list; b; b = (b2VisBody*) b->GetNext()) {
         if(b == this) {
+            setUniqueId(index);
```

Comment:
Would this still cause problems with object IDs in variations of the scene? Because we are still setting unique IDs based on the indexes of the bodies.

Reply:
Okay, because we already do set index after reading from JSON, so this solves the problem.